#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'cuboid_bilateral_symmetric_segmentation'

try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *;
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *;

from math import pi

gen = ParameterGenerator ()

gen.add("color_importance", double_t, 0, "color importance", 0.2, 0.0, 5.0)
gen.add("spatial_importance", double_t, 0, "spatial importance", 4.6, 0.0, 5.0)
gen.add("normal_importance", double_t, 0, "normal importance", 5.0, 0.0, 5.0)
gen.add("use_transform", bool_t, 0, "use transform", True)
gen.add("seed_resolution", double_t, 0, "seed resolution", 0.07, 0.0, 10.0)
gen.add("voxel_resolution", double_t, 0, "voxel resolution", 0.004, 0.0, 1.0)
gen.add("coplanar_threshold", double_t, 0, "coplanar_threshold", 0.00, -5.0, 5.0)
gen.add("distance_threshold", double_t, 0, "distance_threshold", 0.05, 0.0, 10.0)
gen.add("angle_threshold", double_t, 0, "angle_threshold", 30, 0.0, 360.0)
exit (gen.generate (PACKAGE, "cuboid_bilateral_symmetric_segmentation", "SupervoxelSegmentation"))
